lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -2062,10 +2071,31 @@ struct FoldResetMux : public mlir::RewritePattern {
 };
 } // namespace
 
+static bool isDefinedByOneConstantOp(Value v) {
+  if (auto c = v.getDefiningOp<ConstantOp>())
+    return c.getValue().isOne();
+  if (auto sc = v.getDefiningOp<SpecialConstantOp>())
+    return sc.getValue();
+  return false;
+}
+
+static LogicalResult
+canonicalizeRegResetWithOneReset(RegResetOp reg, PatternRewriter &rewriter) {
```

Comment:
This is where `RegResetWithOneReset` went, couldn't sort how to make it work from ODS so re-implemented it in C++.  Will revisit-- issue is `NativeCodeCall` only allows returning a single result, but perhaps it can be reworked to do what we need so we can keep this in ODS!

Reply:
Hmm, can't seem to make this work in ODS.  Can ODS forceable declarations (`RegResetWithZeroReset` works), but not sure if there's a way to get the `DropWrite` functionality needed here, in my attempts it always considers it a single result regardless.

Reply:
Ah sweet, I didn't make the connection between the removed ODS thingy and this explicit canonicalizer.